Initial work on Rook operator Helm Chart

hunter opened the pull request:
This chart launches the Rook operator with options to disable RBAC.

Inspired by the [Prometheus Operator](https://github.com/coreos/prometheus-operator/tree/master/helm/prometheus-operator)

More work to be done with a Cluster chart.

hunter commented on demo/helm/rook-operator/README.md:
```diff
@@ -0,0 +1,110 @@
+rook-operator
```
I've not had a chance to test. I thought I'd get this out first for feedback

luxas replied:
I thought we had agreed to not support v1.5... @bassam

hunter commented on demo/helm/rook-operator/README.md:
```diff
@@ -0,0 +1,110 @@
+
+Installs [rook-operator](https://github.com/rook/rook) to create/configure/manage Rook clusters atop Kubernetes.
+
+TL;DR;
```
The README was just a search and replace from some of the other charts. I'd not noticed that but will update

hunter commented on demo/helm/rook-operator/README.md:
```diff
@@ -0,0 +1,110 @@
+
+If role-based access control (RBAC) is enabled in your cluster, you may need to give Tiller (the server-side component of Helm) additional permissions. **If RBAC is not enabled, be sure to set `rbacEnable` to `false` when installing the chart.**
+
+1.	Create a ServiceAccount for Tiller in the `kube-system` namespace
```
Likely my markdown formatter mixing things up. Will update

hunter commented on demo/helm/rook-operator/README.md:
````diff
@@ -0,0 +1,110 @@
+To uninstall/delete the `rook-operator` deployment:
+
+```console
+$ helm delete rook-operator
````
That's correct. It will only delete the resources specified in the chart. Does the operator launch anything from its deployment?

I've started a second chart for the cluster TPR so that will need to do more cleanup

hunter commented on demo/helm/rook-operator/README.md:
```diff
@@ -0,0 +1,110 @@
+Configuration
+-------------
+
+The following tables lists the configurable parameters of the prometheus-operator chart and their default values.
```
Whoops :)

hunter commented on demo/helm/rook-operator/values.yaml:
```diff
@@ -0,0 +1,20 @@
+
+resources:
+  limits:
+    cpu: 100m
```
Yes, that is the chart generator default. Does it seem in the right ballpark for the operator?

hunter commented on demo/helm/rook-operator/Chart.yaml:
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+description: A Helm chart for Kubernetes
+name: rook-operator
+version: 0.1.0
```
The convention for Helm charts is different to the version numbering of the app/service in the chart. I found it a little confusing at first but it does allow for the development of the chart independently from the app/service version. That way a new value/resource/etc can be added to the chart without having to change the app version, rebuild container images, etc.

jefflaplante replied:
I took @hunter's PR and made a stand alone helm chart out of it. https://github.com/jefflaplante/helm-rook-operator

hunter commented on demo/helm/rook-operator/templates/deployment.yaml:
```diff
@@ -0,0 +1,29 @@
+        args: ["operator"]
+        env:
+        - name: ROOKD_REPO_PREFIX
+          value: quay.io/rook
```
I quickly added a prefix value. Can change or remove if the operator gets an update.

hunter commented:
I was planning on raising a second PR with the cluster TPR chart since it requires more work and testing but can include it in this if you'd prefer

luxas commented on demo/helm/rook-operator/README.md:
```diff
@@ -0,0 +1,109 @@
+Prerequisites
+-------------
+
+- Kubernetes 1.5+ with Beta APIs & ThirdPartyResources enabled
```
Kubernetes v1.6+

luxas commented on demo/helm/rook-operator/templates/deployment.yaml:
```diff
@@ -0,0 +1,29 @@
+apiVersion: apps/v1beta1
```
This is a k8s v1.6+ API group (and that's fine, just pointing out)

jefflaplante commented:
I took @hunter's PR and made a stand alone helm chart out of it. https://github.com/jefflaplante/helm-rook-operator

jefflaplante commented:
@travisn - It's temporary until the PR is merged. However, you'll want to create a helm chart repository to make it easily accessible to helm users.

```
cd demo/helm/rook-operator
helm package .
helm repo index --url helm.rook.io .
upload generated index.yaml and *.tgz to wherever helm.rook.io is hosted.
```
I've added it to my helm repository here: http://helm.jefflaplante.com for the time being.

jefflaplante commented:
Helm users can then use the chart by adding your chart repository.

```
$ helm repo add rook http://helm.rook.io
$ helm repo update
$ helm search rook
NAME                      	VERSION	DESCRIPTION
rook/rook-operator	0.1.0  	A Helm chart for a Rook operator
$ helm install rook/rook-operator -n rook --namespace rook
```

DanKerns commented:
We should have an issue to track this…

jefflaplante commented:
https://github.com/rook/rook/issues/284

This issue is tracking helm install of rook.

jefflaplante commented:
@travisn - I believe I fixed the issue you're having in my branch.

```
$ helm repo add jefflaplante http://helm.jefflaplante.com
"jefflaplante" has been added to your repositories

$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "jefflaplante" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈

$ helm install jefflaplante/rook-operator -n rook-operator
NAME:   rook-operator
LAST DEPLOYED: Mon Jun  5 18:54:18 2017
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/ClusterRole
NAME           AGE
rook-operator  1s

==> v1beta1/ClusterRoleBinding
NAME           AGE
rook-operator  1s

==> v1beta1/Deployment
NAME           DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
rook-operator  1        1        1           0          1s

==> v1/ServiceAccount
NAME           SECRETS  AGE
rook-operator  1        1s
```

As for the helm repo hosting - I'm doing it with a Google Cloud Storage bucket. It could just as well as be any other static hosting - AWS bucket or github repo.

bassam commented:
@jefflaplante thanks. #693 is tracking publishing helm charts on each release.

hunter commented:
Strange. I thought I'd fixed the issue with ingress. Let me check what happened

hunter commented:
@travisn I can't replicate. Could you be running with older commits or using a helm chart generated from the previous code?

I've tested the deployment of this operator. All seems to be working as expected. I've got a cluster chart that I've started work on but this is ready to go unless you want to hold out until its ready

hunter commented:
@travisn I've rebased and squashed the commits. Agreed about support for only 1.6, I've removed any references to 1.5 in the README and also gotten rid of the conditional in the template for the `apiVersion`.

I've also added some info to the TL;DR section on how to deploy the chart locally

rimusz commented:
👍
